include/interfaces/turbulence_k_omega.h

```diff
@@ -0,0 +1,708 @@
+/*! \addtogroup equations
+ *  @{
+ */
+
+/**
+ *  see Wilcox, D.C. (1988), "Re-assessment of the scale-determining
+ *  equation for advanced turbulence models", AIAA Journal, vol. 26,
+ *  no. 11, pp. 1299-1310.
+ */
+
+#ifndef _pidomus_turbulence_k_omega_h_
+#define _pidomus_turbulence_k_omega_h_
+
+#include "pde_system_interface.h"
+
+#include <deal.II/lac/linear_operator.h>
+#include <deal.II/lac/block_linear_operator.h>
+#include <deal.II/lac/packaged_operation.h>
+#include <deal.II/lac/solver_cg.h>
+#include <deal.II/lac/solver_gmres.h>
+
+template <int dim, int spacedim=dim, typename LAC=LATrilinos>
+class KOmega
+  :
+  public PDESystemInterface<dim,spacedim,KOmega<dim,spacedim,LAC>, LAC>
+{
+
+public:
+  ~KOmega () {};
+  KOmega ();
+
+  void declare_parameters (ParameterHandler &prm);
+  void parse_parameters_call_back ();
+
+  template <typename EnergyType, typename ResidualType>
+  void
+  energies_and_residuals(
+    const typename DoFHandler<dim,spacedim>::active_cell_iterator &cell,
+    FEValuesCache<dim,spacedim> &scratch,
+    std::vector<EnergyType> &energies,
+    std::vector<std::vector<ResidualType>> &residuals,
+    bool compute_only_system_terms) const;
+
+  void
+  compute_system_operators(
+    const std::vector<shared_ptr<LATrilinos::BlockMatrix>>,
+    LinearOperator<LATrilinos::VectorType> &,
+    LinearOperator<LATrilinos::VectorType> &,
+    LinearOperator<LATrilinos::VectorType> &) const;
+
+  void
+  set_matrix_couplings(std::vector<std::string> &couplings) const;
+
+private:
+  /**
+   * Enable dynamic term: \f$ \partial_t u\f$.
+   */
+  bool dynamic;
+
+  /**
+   * Enable convection term: \f$ (\nabla u)u \f$.
+   * This term introduce the non-linearity of Navier Stokes Equations.
+   */
+  bool convection;
+
+  /**
+   * Hot to handle \f$ (\nabla u)u \f$.
+   */
+  std::string non_linear_term;
+
+  /**
+   * Hot to handle \f$ (\nabla u)u \f$.
+   */
+  bool linearize_in_time;
+
+  /**
+  * Name of the preconditioner:
+  */
+  std::string prec_name;
+
+  /**
+   * Density
+   */
+  double rho;
+
+  /**
+   * Viscosity
+   */
+  double nu;
+
+  /**
+   * Viscosity
+   */
+  mutable double nu_T;
+
+  /**
+   *
```

Comment:
Please, add a documentation of each of the parameters, maybe pointing to the equation name in the paper